File: app/src/main/java/com/volkanhotur/basemvvm/android/data/repository/ApiDataFactory.java
package com.volkanhotur.basemvvm.android.data.repository;

import java.util.HashMap;

import javax.inject.Inject;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.Response;

/**
 * @author volkanhotur
 */

public class ApiDataFactory implements Repository {

    private ApiDataSource source;

    @Inject
    public ApiDataFactory(ApiDataSource source) {
        this.source = source;
    }

    @Override
    public Observable<ResponseBody> postRequest(String url, HashMap<String, Object> params, HashMap<String, Object> headers) {
        return source.postRequest(url, params, headers);
    }

    @Override
    public Observable<ResponseBody> postRequest(String url, HashMap<String, Object> params) {
        return source.postRequest(url, params);
    }

    @Override
    public Observable<ResponseBody> getRequest(String url, HashMap<String, Object> params, HashMap<String, Object> headers) {
        return source.getRequest(url, params, headers);
    }

    @Override
    public Observable<ResponseBody> getRequest(String url) {
        return source.getRequest(url);
    }
}

File: app/src/main/java/com/volkanhotur/basemvvm/android/data/repository/ApiDataFactory.kt
package com.volkanhotur.basemvvm.android.data.repository

import io.reactivex.Observable
import okhttp3.ResponseBody
import java.util.*
import javax.inject.Inject

/**
 * @author volkanhotur
 */
class ApiDataFactory @Inject constructor(private val source: ApiDataSource) : Repository {

    override fun postRequest(url: String?, params: HashMap<String?, Any?>?, headers: HashMap<String?, Any?>?): Observable<ResponseBody?> {
        return source.postRequest(url, params, headers)
    }

    override fun postRequest(url: String?, params: HashMap<String?, Any?>?): Observable<ResponseBody?> {
        return source.postRequest(url, params)
    }

    override fun getRequest(url: String?, params: HashMap<String?, Any?>?, headers: HashMap<String?, Any?>?): Observable<ResponseBody?> {
        return source.getRequest(url, params, headers)
    }

    override fun getRequest(url: String?): Observable<ResponseBody?> {
        return source.getRequest(url)
    }
}